app/src/main/java/com/github/sdp_begreen/begreen/activities/SignInActivity.kt

```diff
@@ -93,13 +99,43 @@ class SignInActivity : AppCompatActivity() {
                     if (it.isNotBlank()) {
                         checkUserExistence()
                         startActivity(Intent(this, MainActivity::class.java))
-                        // Here I will hide the progress bar when implemented
+                        hideProgressDialog()
                         finish()
                     }
                 }
             }
     }
 
+    // Variable to hold the progress dialog
+    private var dialog: AlertDialog? = null
```

Comment:
You can move up this variable to the top of the class so all variables are declared in the same place

Reply:
Well spotted, thanks